sunpy/map/compositemap.py

```diff
@@ -439,8 +439,12 @@ def plot(self, axes=None, annotate=True,
             # non False levels property.
             if m.levels is False:
                 # We tell GenericMap.plot() that we need to autoalign the map
-                if wcsaxes_compat.is_wcsaxes(axes):
+                
+                if wcsaxes_compat.gca_wcs(m.wcs)==axes.wcs:
+                    params['autoalign'] = False
+                elif wcsaxes_compat.is_wcsaxes(axes):
                     params['autoalign'] = True
```

Comment:
How does this speed up the plotting? I am not familiar with CompositeMap plotting

Reply:
See #5894.  We get improved performance if we set `autoalign=False` when the map layer is being plotted onto axes with a matching WCS because then we use the faster `imshow()` instead of the slower `pcolormesh()`.

Reply:
Ah, so the idea is to bypass using pcolormesh when it makes sense?

Reply:
Yup, precisely, because it's glacial in an interactive plot.